lib/galaxy/datatypes/registry.py

```diff
@@ -127,8 +133,8 @@ def __import_module(full_path: str, datatype_module: str):
             return module
 
         if root_dir and config:
-            compressed_sniffers = {}
-            if isinstance(config, (str, Path)):
+            compressed_sniffers: Dict[Type[Data], List[Data]] = {}
+            if isinstance(config, (str, os.PathLike)):
```

Comment:
I wonder if this would work :thinking: 
```suggestion
            if isinstance(config, StrPath):
```

Reply:
No, unfortunately, because it's defined as `StrPath = Union[str, PathLike]` and `isinstance()` supports Union types only from Python 3.10, see https://docs.python.org/3/library/functions.html#isinstance .

Reply:
Ahh right, we still support lower Python versions :sweat_smile: all right!

Reply:
Even if we supported only Python >=3.10, this would fail type checking with mypy:
```
lib/galaxy/datatypes/registry.py:137: error: Parameterized generics cannot be used with class or instance checks  [misc]
                if isinstance(config, StrPath):
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
lib/galaxy/datatypes/registry.py:137: error: Argument 2 to "isinstance" has incompatible type "<typing special form>"; expected "_ClassInfo"  [arg-type]
                if isinstance(config, StrPath):
                                      ^~~~~~~
```

because we are defining it as:

```python
if TYPE_CHECKING:
    StrPath = Union[str, PathLike[str]]

else:
    StrPath = Union[str, PathLike]
```

Reply:
Ahh, I see! Thanks for checking in detail!